app/main/views/services.py

```diff
@@ -507,6 +508,74 @@ def update_section_submission(framework_slug, lot_slug, service_id, section_id,
                                 _anchor=section_id))
 
 
+@main.route('/frameworks/<framework_slug>/submissions/<lot_slug>/<service_id>/remove/<section_id>/<question_slug>',
+            methods=['GET', 'POST'])
+@login_required
+def remove_subsection(framework_slug, lot_slug, service_id, section_id, question_slug):
+    try:
+        draft = data_api_client.get_draft_service(service_id)['services']
+    except HTTPError as e:
+        abort(e.status_code)
+
+    if not is_service_associated_with_supplier(draft):
+        abort(404)
+
+    content = content_loader.get_manifest(framework_slug, 'edit_submission').filter(draft)
+    question_to_remove = content.get_question_by_slug(question_slug)
+    containing_section = content.get_section(section_id)
```

Comment:
This should be in the opposite order: find the section, get `section.get_question_by_slug` if the is not None. This is also missing the `editable` checks, which means it could make it possible to delete read-only fields (I think the API will complain, but I think this could still be a problem). I'd copy the relevant bit from `edit_service_submission` since that's likely to do the right thing.